components/challenges/TimerClock.js

```diff
@@ -4,11 +4,35 @@ import ProgressCircle from 'react-native-progress-circle';
 
 import Colors from "../../constants/Colors";
 
+import { getUserId } from "../../utils/auth/auth";
+
 import KitText from "../KitText"
 const diameter = 96;
 export default function TimerClock(props) {
   function calculatePercent() {
-    return 35;
+    const minutesGiven = props.challenge.challengeDetails.minGiven;
+    const opened = props.challenge.opened || null; //In theory we should never pass an assignedChallenge to the TimerClock with the user not in the submissions array
+
+    if (!opened) return 35; //EVAN TODO: this should be an error
+    const currentUser = getUserId();
+    const timeOpened = opened[currentUser]
+    const currentTime = new Date().getTime() / 1000; //Adjusting to seconds
+  
+    // console.log((currentTime - timeOpened)/60)
+    // console.log((currentTime - timeOpened) / (minutesGiven * 60))
+    return (currentTime - timeOpened) / (minutesGiven * 60) * 100; //Converting minutes to seconds, then converting decimal to percent
+  }
+
+  function calculateHoursLeft() {
```

Comment:
Not blocking but I made a time.js in utils in another branch and this should eventually be moved there I think.